crates/net/dns/src/lib.rs

```diff
@@ -392,22 +390,33 @@ pub enum DnsDiscoveryEvent {
 
 /// Converts an [Enr] into a [NodeRecord]
 fn convert_enr_node_record(enr: Enr<SecretKey>) -> Option<DnsNodeRecordUpdate> {
-    use alloy_rlp::Decodable;
-
     let node_record = match NodeRecord::try_from(&enr) {
         Ok(node_record) => node_record,
         Err(err) => {
             trace!(target: "disc::dns",
                 %err,
-                "can't convert enr to node_record"
+                "can't convert enr to dns update"
             );
 
             return None
         }
     };
 
-    let mut maybe_fork_id = enr.get(b"eth")?;
-    let fork_id = ForkId::decode(&mut maybe_fork_id).ok();
+    let fork_id = match get_fork_id(&enr) {
+        Ok(fork_id) => Some(fork_id),
+        Err(err) => {
+            if matches!(err, NodeRecordParseError::EthForkIdMissing) {
+                trace!(target: "disc::dns",
+                    %err,
+                    "can't convert enr to dns update"
+                );
+
+                return None
+            }
+
+            None // enr fork could be badly configured in node record, peer could still be useful
+        }
```

Comment:
hmm, can we not keep what we had previously?

Reply:
sure, am thinking of other pr where I use this method again

Reply:
(informative errors are always helpful)

Reply:
reverted now, not letting this detail be a blocker, but would like to get input from @Rjected here?

Reply:
Let's move this discussion and change to the other PR if this is only relevant there. If we want a trace when a dns enr doesn't have an `eth` entry, we could add a trace in an if block or something. Not sure that this warrants another method, let's see the reuse in the other PR